replica/table.cc

```diff
@@ -1838,6 +1840,33 @@ void table::update_effective_replication_map(locator::effective_replication_map_
     if (old_erm) {
         old_erm->invalidate();
     }
+
+    _stats.tablet_count_on_current_shard = calculate_tablet_count_on_current_shard();
+}
+
+int64_t table::calculate_tablet_count_on_current_shard() const {
```

Comment:
Is this calculation done per table?

Reply:
In line 1853 tablet_map is retrieved as follows:
`const auto& tablet_map = token_metadata.tablets().get_tablet_map(_schema->id());`

In `./locator/tablets.hh` I can see that `tablet_metadata::get_tablet_map()` takes `table_id` as the parameter.
`const tablet_map& get_tablet_map(table_id id) const;`

The implementation of this member function is as follows:
```
const tablet_map& tablet_metadata::get_tablet_map(table_id id) const {
    try { 
        return _tablets.at(id);
    } catch (const std::out_of_range&) {
        throw_with_backtrace<std::runtime_error>(format("Tablet map not found for table {}", id));
    } 
}
```

So, the answer for this question seems to be yes - the calculation is done per table.

Is it sufficient? Does it need to change?